FindSTM32G4xx.cmake

```diff
@@ -103,6 +103,15 @@ set(STM32G4xx_BSP_VERSION "1.4.0" PARENT_SCOPE)
 set(STM32G4xx_BSP_VERSION_STRING "1.4.0" PARENT_SCOPE)
 
 file(GLOB_RECURSE hal_driver_sources ${bsp_source}/Drivers/STM32G4xx_HAL_Driver/Src/*.c)
+# GLOB_RECURSE lists files in alphabetical order. That means that the hal, which is
+# structured so that each subsystem has stm32g4xx_hal_subsystem.c and
+# stm32g4xx_hal_subsystem_ex.c, will have subsystem.c packed into the archive
+# before hal_subsystem_ex.c. That breaks some of the hal code, which has weak definitions
```

Comment:
Is it always guaranteed to be in reverse order?

Reply:
`GLOB_RECURSE` returns results sorted lexicographically since 3.6 per https://cmake.org/cmake/help/latest/command/file.html#glob-recurse yes